src/librustc_borrowck/diagnostics.rs

```diff
@@ -314,6 +314,59 @@ let c = &i; // still ok!
 ```
 "##,
 
+E0504: r##"
+This error occurs when an attempt is made to move a borrowed variable into a
+closure. For example:
+
+```compile_fail
```

Comment:
No need to use threads here. It shadows a bit the explanation and makes the example harder than necessary. You could use:

```Rust
struct FancyNum {
    num: u8
}

fn main() {
    let fancy_num = FancyNum { num: 5 };
    let fancy_ref = &fancy_num;

    let x = move || {
        println!("child thread: {}", fancy_num.num);
        // error: cannot move `fancy_num` into closure because it is borrowed
    };

    println!("main thread: {}", fancy_ref.num);
}
```